admin/server/organizations.go

```diff
@@ -415,52 +415,64 @@ func (s *Server) RemoveOrganizationMember(ctx context.Context, req *adminv1.Remo
 		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
 
-	claims := auth.GetClaims(ctx)
-	if !claims.OrganizationPermissions(ctx, org.ID).ManageOrgMembers {
-		return nil, status.Error(codes.PermissionDenied, "not allowed to remove org members")
-	}
-
 	user, err := s.admin.DB.FindUserByEmail(ctx, req.Email)
 	if err != nil {
 		if !errors.Is(err, database.ErrNotFound) {
 			return nil, status.Error(codes.Internal, err.Error())
 		}
-		// check if there is a pending invite
+
+		// Only admins can remove pending invites.
+		// NOTE: If we change invites to accept/decline (instead of auto-accept on signup), we need to revisit this.
+		claims := auth.GetClaims(ctx)
+		if !claims.OrganizationPermissions(ctx, org.ID).ManageOrgMembers {
+			return nil, status.Error(codes.PermissionDenied, "not allowed to remove org members")
+		}
+
+		// Check if there is a pending invite
 		invite, err := s.admin.DB.FindOrganizationInvite(ctx, org.ID, req.Email)
 		if err != nil {
 			if errors.Is(err, database.ErrNotFound) {
 				return nil, status.Error(codes.InvalidArgument, "user not found")
 			}
 			return nil, status.Error(codes.Internal, err.Error())
 		}
+
 		err = s.admin.DB.DeleteOrganizationInvite(ctx, invite.ID)
 		if err != nil {
 			return nil, status.Error(codes.Internal, err.Error())
 		}
+
 		return &adminv1.RemoveOrganizationMemberResponse{}, nil
 	}
 
+	// The caller must either have ManageOrgMembers permission or be the user being removed.
+	claims := auth.GetClaims(ctx)
+	allow := false
+	allow = allow || claims.OrganizationPermissions(ctx, org.ID).ManageOrgMembers
+	allow = allow || (claims.OwnerType() == auth.OwnerTypeUser && claims.OwnerID() == user.ID)
```

Comment:
```suggestion
	allow := claims.OrganizationPermissions(ctx, org.ID).ManageOrgMembers || (claims.OwnerType() == auth.OwnerTypeUser && claims.OwnerID() == user.ID)
```

Reply:
See my response on the other review comment